pouta_blueprints/drivers/provisioning/openshift_driver.py

```diff
@@ -0,0 +1,638 @@
+                                        'port': port,
+                                        'scheme': 'HTTP',
+
+                                    }
+                                }
+                            }
+                        ],
+                    }
+                }
+            }
+        }
+
+        dc_volume_mounts = []
+        dc_volumes = []
+
+        # workaround for service account secret automount.
+        # see https://github.com/kubernetes/kubernetes/issues/16779
+        # TODO revise this workaround
+        dc_volume_mounts.append(
+            {
+                'mountPath': '/var/run/secrets/kubernetes.io/serviceaccount',
+                'name': 'nosecret',
+            }
+        )
+        dc_volumes.append(
+            {
+                'name': 'nosecret',
+                'emptyDir': {}
+            }
+        )
+
+        # add a mount point to persistent storage, if configured
+        if volume_mount_point:
+            dc_volume_mounts.append(
+                {
+                    'mountPath': volume_mount_point,
+                    'name': 'work',
+                }
+            )
+            dc_volumes.append(
+                {
+                    'name': 'work',
+                    'persistentVolumeClaim': {
+                        'claimName': 'pvc001'
+                    }
+                }
+            )
+
+        dc_data['spec']['template']['spec']['containers'][0]['volumeMounts'] = dc_volume_mounts
+        dc_data['spec']['template']['spec']['volumes'] = dc_volumes
+
+        # create environment variables for the blueprint
+        env_data = []
+        if environment_vars:
+            for key, value in environment_vars.items():
+                env_data.append({'name': key, 'value': value})
+
+        dc_data['spec']['template']['spec']['containers'][0]['env'] = env_data
+
+        oc.make_request(namespace=project_name, object_kind='deploymentconfigs', data=dc_data)
+
+        pod_ready = False
+        # wait for pod to become ready
+        for i in range(120):
+            # pods live in kubernetes API
+            res = oc.make_request(
+                kubeapi=True,
+                namespace=project_name,
+                object_kind='pods',
+                params=dict(labelSelector='run=%s' % pod_name),
+            )
+
+            if res.ok:
+                pod_ready = None
+                for pod in res.json()['items']:
+                    try:
+                        for containerStatus in pod['status']['containerStatuses']:
+                            if containerStatus['name'] == pod_name:
+                                pod_ready = containerStatus['ready']
+                                break
+                    except:
+                        pass
+
+                if pod_ready:
+                    break
+
+            self.logger.debug('waiting for pod to be ready %s' % pod_name)
+            time.sleep(5)
+
```

Comment:
I did not even know of while...else construct. Not sure if I'm completely sold, but giving it a go anyway. Thanks!